### ISSUE TITLE
[Bug]: Different vector dimensions

### ISSUE THREAD

**Issue by bidoubiwa** [QuivrHQ/quivr#2263]
### What happened?

I'm trying to run quivr with Ollama/mistral. After a search, the request goes through to Ollama correctly and returns a 200. 

```
[GIN] 2024/02/24 - 20:45:46 | 200 |  270.057583ms |       127.0.0.1 | POST     "/api/embeddings"
```

After which, a bug occurs in the back-end with postgres.

```
backend-core  | postgrest.exceptions.APIError: {'code': '22000', 'details': None, 'hint': None, 'message': 'different vector dimensions 1536 and 4096'}
```

I found something about postgres vector size error in this issue: https://github.com/langchain-ai/langchain/issues/2219 but wanted your opinion before.

### Relevant log output

```bash
backend-core  | INFO:     172.25.0.1:57088 - "POST /chat/dfc971ff-da3f-4ca2-a305-ae4c5b554d23/question/stream?brain_id=8e04a954-97c0-41ef-89e5-e05fd5c59703 HTTP/1.1" 500 Internal Server Error
backend-core  | ERROR:    Exception in ASGI application
backend-core  | Traceback (most recent call last):
backend-core  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
backend-core  |     result = await app(  # type: ignore[func-returns-value]
backend-core  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
backend-core  |     return await self.app(scope, receive, send)
backend-core  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
backend-core  |     await super().__call__(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
backend-core  |     await self.middleware_stack(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
backend-core  |     raise exc
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
backend-core  |     await self.app(scope, receive, _send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 91, in __call__
backend-core  |     await self.simple_response(scope, receive, send, request_headers=headers)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 146, in simple_response
backend-core  |     await self.app(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
backend-core  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
backend-core  |     raise exc
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
backend-core  |     await app(scope, receive, sender)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 758, in __call__
backend-core  |     await self.middleware_stack(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 778, in app
backend-core  |     await route.handle(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 299, in handle
backend-core  |     await self.app(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 79, in app
backend-core  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
backend-core  |     raise exc
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
backend-core  |     await app(scope, receive, sender)
backend-core  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 74, in app
backend-core  |     response = await func(request)
backend-core  |                ^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 299, in app
backend-core  |     raise e
backend-core  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 294, in app
backend-core  |     raw_response = await run_endpoint_function(
backend-core  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
backend-core  |     return await dependant.call(**values)
backend-core  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/code/modules/chat/controller/chat_routes.py", line 232, in create_stream_question_handler
backend-core  |     gpt_answer_generator = get_answer_generator(
backend-core  |                            ^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/code/modules/chat/controller/chat_routes.py", line 76, in get_answer_generator
backend-core  |     brain, metadata_brain = brain_service.find_brain_from_question(
backend-core  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/code/modules/brain/service/brain_service.py", line 109, in find_brain_from_question
backend-core  |     list_brains = vector_store.find_brain_closest_query(user.id, question)
backend-core  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
backend-core  |   File "/code/vectorstore/supabase.py", line 54, in find_brain_closest_query
backend-core  |     ).execute()
backend-core  |       ^^^^^^^^^
backend-core  |   File "/usr/local/lib/python3.11/site-packages/postgrest/_sync/request_builder.py", line 119, in execute
backend-core  |     raise APIError(r.json())
backend-core  | postgrest.exceptions.APIError: {'code': '22000', 'details': None, 'hint': None, 'message': 'different vector dimensions 1536 and 4096'}
backend-core  | 2024-02-24 19:45:58,242:INFO - HTTP Request: GET http://host.docker.internal:54321/rest/v1/onboardings?select=onboarding_a%2Conboarding_b1%2Conboarding_b2%2Conboarding_b3&user_id=eq.39418e3b-0258-4452-af60-7acfcc1263ff&limit=1 "HTTP/1.1 200 OK"
```

**Comment by StanGirard** [comment]
Hey my friend, you can't have multiple dimensions vectors in the same database. Please delete all of your vectors and change the dimensions of your vectors

**Comment by bidoubiwa** [comment]
Hey @StanGirard,

Thanks for your help :) Unfortunately I still get the error. This is what I did: 

## Steps
- Changed the vector size in the sql migrations files:  `"embedding" vector(1536)` =>  `"embedding" vector(4096)`
- Changed the default model to `["ollama/mistral"]` in the migration files.
- Added the model and the api_base_url directly in the BrainSettings
```py
   class BrainSettings(BaseSettings):
    # ....
    ollama_api_base_url: str = "http://host.docker.internal:11434"
    model: str = "mistral"
   # ...
```
```py
# in get_embeddings
embeddings = OllamaEmbeddings(
  base_url=settings.ollama_api_base_url,
  model=settings.model
```
- Reset the supabase db => `supabase db reset`. Just to be sure, I also deleted the docker volumes

## Chat Error

Whenever I try to chat using the Quivr dashboard the same error is thrown. 

The call to ollama is succesfull though: 
```
[GIN] 2024/02/26 - 17:43:44 | 200 |  6.618862708s |       127.0.0.1 | POST     "/api/embeddings"
```

From what I see the error is triggered when the `/stream` route is called
```
backend-core  | INFO:     172.31.0.1:59116 - "POST /chat/b5be5c51-7cec-4816-ab9c-4dc4625e041d/question/stream HTTP/1.1" 500 Internal Server Error
```

The method that throws is `find_brain_closest_query` in `supabase.py` when the `execute()` is triggered.

Which in the front-end console translates to: 
```
Access to fetch at 'http://localhost:5050/chat/b5be5c51-7cec-4816-ab9c-4dc4625e041d/question/stream' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

I checked the tables afterwards and the vector table is still empty.

**Comment by StanGirard** [comment]
Ah thanks ! 

Ok there is another table called brains with vectors (meaning) in it. Update this one as well ;) 

We vectorize the brains also to find the closest one to a query

**Comment by bidoubiwa** [comment]
Good news, I updated the meaning with the embedding of `Default brain`, I'm not having the same error anymore.
Though, now I'm getting this error

> | litellm.exceptions.AuthenticationError: OpenAIException - Error code: 401 - {'error': {'message': 'Incorrect API key provided: CHANGE_ME. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}

While my user correctly has `mistral` as its model

<img width="1409" alt="Screenshot 2024-02-26 at 18 54 41" src="https://github.com/QuivrHQ/quivr/assets/33010418/de0e1e68-3d96-4877-ac92-fc08229ce2c5">

Nonetheless! This is another issue :) This one is fixed <3

**Comment by aomeng1219** [comment]
[ERROR] packages.embeddings.vectors [vectors.py:23]: Error creating vector for document {'code': '22000', 'details': None, 'hint': None, 'message': 'expected 1536 dimensions, not 4096'}
Have you solved this problem? I have also encountered this problem, and this error still exists regardless of switching any model, whether it is ollama or openAI.
